src/pyhf/infer/intervals.py

```diff
@@ -0,0 +1,39 @@
+from . import hypotest
+from .. import get_backend
+
+
+def _interp(x, xp, fp):
+    import numpy as np
+
+    tb, _ = get_backend()
+    return tb.astensor(np.interp(x, xp.tolist(), fp.tolist()))
+
+
+def upperlimit(data, model, scan, return_results=False):
```

Comment:
Given my comment in the PR, it would be good to set an `alpha` (or similar) API for setting the upper limit
```suggestion
def upperlimit(data, model, scan, alpha=0.05, return_results=False):
```

so the below would use

```python
limits = [_interp(alpha, resarary[i][::-1], scan[::-1]) for i in range(6)]
```

Reply:
hm we als use hypotest and not hypo_test. Thoughts?

Reply:
This is true. I've been tempted to change `hypotest`, except that I think a lot of people already are relying explicitly on that API so I'm hesitant to change it now. `hypotest` and `upperlimit` is more in the style of the [NumPy API](https://numpy.org/doc/1.18/reference/arrays.ndarray.html) choices, while `upper_limit` is more in the style of the [SciPy](https://docs.scipy.org/doc/scipy/reference/linalg.html), [Matplotlib](https://matplotlib.org/api/index.html#usage-patterns), and [PyTorch](https://pytorch.org/docs/stable/tensors.html) API style choices. I'm personally more partial to the latter, but that's just a _preference_.

Reply:
@kratsg also prefers `hypotest` and `upperlimit` so let's go with that.